core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java

```diff
@@ -73,8 +73,14 @@
             // 2. initiate RudderElementCache
             RudderLogger.logDebug("EventRepository: constructor: Initiating RudderElementCache");
             RudderElementCache.initiate(_application);
+            preferenceManager = RudderPreferenceManager.getInstance(_application);
+            String anonymousId;
+            RudderLogger.logDebug(RudderElementCache.getCachedContext().getAnonymousId());
+            if (RudderElementCache.getCachedContext().getAnonymousId() == null)
```

Comment:
Ideally, once you have the the `RudderElementCache` ready you should have `anonymousId` as well and you should be able to take directly from there without bothering about null value

Reply:
done